numpy/ma/core.py

```diff
@@ -3316,6 +3317,12 @@ def _scalar_heuristic(arr, elem):
                                 f"all to {dout._fill_value[0]!s}.",
                                 stacklevel=2)
                         dout._fill_value = dout._fill_value.flat[0]
+                    elif not hasattr(dout._fill_value, 'ndim'):
+                        # only initialize to empty array if it dout does not 
+                        # have ._fill_value before. If it has, do not change 
+                        # the original behavior
+                        dout._fill_value = np.empty((), dtype)
+                        dout._fill_value[()] = self._fill_value[indx]
```

Comment:
I suspect that actually the fix is way simpler, and that you can get away with the only change being
```suggestion
                        # need to use squeeze to ensure the result is an array not a scalar
                        dout._fill_value = dout._fill_value.flat[0:1].squeeze(axis=0)
```

Would you mind testing that?

Reply:
Oh yes I've tested it and it works! Thank you!

Reply:
I'll pushed the tests along with the change

Reply:
Thanks for taking the time to write the tests so that it's easy to try these things!